doc/sphinx/proofs/writing-proofs/inductive.rst

```diff
@@ -348,40 +352,38 @@ analysis on inductive or co-inductive objects (see :ref:`inductive-definitions`)
          induction y in x |-   *.
          Show 2.
 
-.. tacv:: induction @term with @bindings as @or_and_intropattern_loc using @term with @bindings in @goal_occurrences
-          einduction @term with @bindings as @or_and_intropattern_loc using @term with @bindings in @goal_occurrences
+.. tacv:: induction @term with @bindings @as_or_and_ipat using @term with @bindings in @goal_occurrences
+          einduction @term with @bindings @as_or_and_ipat using @term with @bindings in @goal_occurrences
 
    These are the most general forms of :tacn:`induction` and :tacn:`einduction`. It combines the
    effects of the with, as, using, and in clauses.
 
-.. tacv:: elim @term
-   :name: elim
+.. tacn:: elim @one_term_with_bindings {? using @one_term {? with @bindings } }
 
    This is a more basic induction tactic. Again, the type of the argument
-   :n:`@term` must be an inductive type. Then, according to the type of the
+   :n:`@one_term` must be an inductive type. Then, according to the type of the
    goal, the tactic ``elim`` chooses the appropriate destructor and applies it
    as the tactic :tacn:`apply` would do. For instance, if the local context
    contains :g:`n:nat` and the current goal is :g:`T` of type :g:`Prop`, then
    :n:`elim n` is equivalent to :n:`apply nat_ind with (n:=n)`. The tactic
    ``elim`` does not modify the context of the goal, neither introduces the
    induction loading into the context of hypotheses. More generally,
-   :n:`elim @term` also works when the type of :n:`@term` is a statement
+   :n:`elim @one_term` also works when the type of :n:`@one_term` is a statement
    with premises and whose conclusion is inductive. In that case the tactic
-   performs induction on the conclusion of the type of :n:`@term` and leaves the
+   performs induction on the conclusion of the type of :n:`@one_term` and leaves the
    non-dependent premises of the type as subgoals. In the case of dependent
    products, the tactic tries to find an instance for which the elimination
    lemma applies and fails otherwise.
```

Comment:
Seems like more words than are necessary.  The first part seems to be that it does `apply nat_ind with (n:=n)` (a confusing choice of variables!).  But I don't quite get the part starting on new line 370 ("More generally ... performs induction ..."--the "performs induction" seems a bit circular!  Can you elaborate on this a little?

"premises" of A -> B -> C would be A and B?  Then what are the non-dependent premises here?

I want to get this clear because `simple induction` is defined in terms of `elim`.

Reply:
> More generally ... performs induction ..."--the "performs induction" seems a bit circular!

Yes, IIUC, it's just defining this more general case in function of the less general case: "if you call me on something whose type is `A -> B -> C`, where `C` is an inductive type, I will introduce the hypotheses `A` and `B` and act on `C` as I described above".

`A` and `B` in this case are indeed the (non-dependent) premises. Dependent premises would be if there was a `forall`. Recall that `A -> B` is just `forall _ : A, B`. It's called non-dependent because the variable of type `A` does not appear in `B`.

Reply:
How does this look?

![image](https://user-images.githubusercontent.com/1253341/121597064-17e53580-c9f5-11eb-8229-d73c62cccaba.png)

And I suppose `forall x:nat, 1 = 0` is pointless.

Reply:
> `A` and `B` are (non-dependent) premises of `A -> B -> C`

Is the premise the type, or the name?  That is, in `forall x : nat, x = x`, is the premise `x` or is it `nat`?

> `A` is a dependent premise of `A -> A \/ B`

I think this is wrong.  `A` is a dependent premise(?) of something like `forall A : Prop, A`, or `forall A : unit, A = A`.

Reply:
Well, the goal context shows types and `A -> B -> C` is a type, so the premise would be a type, too.  I think it's needlessly confusing to be utterly rigorous about names vs types significand.  In algebra, from "x = 2" you would say "x is a variable", "x is an integer" as well as "x is 2".  "x is a variable equal to 2" is unnatural unless you're introducing the concept of a variable to grade-school student.

> I think this is wrong.

Theo?  I'm just the scribe here.

Reply:
> Theo? I'm just the scribe here.

I don't see anything Theo said that I interpret as claiming that `A` and `B` are dependent in `A -> B -> C`.  Is there some bit of text that he wrote that you're referring to here?

Reply:
I'm trying to guess how to describe things I don't know enough about--that is not at all easy to do.  It's probably better for Théo to respond directly first before making this a 3 way conversation.  But, nonetheless, what text would you give to replace everything I wrote above?

Reply:
How about this?


* `n` is a :term:`dependent premise` of `forall n:nat, n + 0 = n` because
  `n` appears both in the binder of the `forall` and in the quantified statement
  `n + 0 = n`.
  
* `A` and `B` are (non-dependent) :gdef:`premises <premise>` of `A -> B -> C`.
  (Recall that `A -> B` is notation for the term `forall _ : A, B`; the wildcard
  `_` can't be used in the quantified statement.)

Reply:
Indeed, Jason is right here and your last proposal looks better (and more correct) than the former.

Indeed, there is an ambiguity regarding what premise refers to (the term or its type). I think in the case of non-dependent premises, it really means the type, because this gives similar language to logic without dependent types. In the case of dependent premises, maybe we can write "`n : nat` is a dependent premise"?